signal/client/grpc.go

```diff
@@ -322,14 +324,29 @@ func (c *GrpcClient) Send(msg *proto.Message) error {
 		return err
 	}
 
-	ctx, cancel := context.WithTimeout(context.Background(), time.Second*2)
-	defer cancel()
-	_, err = c.realClient.Send(ctx, encryptedMessage)
-	if err != nil {
-		return err
+	attemptTimeout := defaultSendTimeout
+
+	for attempt := 0; attempt < 4; attempt++ {
+		select {
+		case <-c.ctx.Done():
+			return fmt.Errorf("not running message attempt. client context is done: %s", c.ctx.Err())
+		default:
+		}
+		if attempt > 1 {
```

Comment:
if you start the iteration from 1 instead of 0 then this if statement no need to.

Reply:
we decided to have the following attempts timeout:
1st and 2nd with the default value 5s
3rd with 10s
4th with 15s